### PR TITLE
Inform student of leaderboard position in level reached popup

### PR DESCRIPTION
**Pull request by kxmbrian**
Student is on leaderboard:
screen shot 2018-02-05 at 2 54 50 pm
Student is not on leaderboard:
screen shot 2018-02-05 at 2 55 07 pm
Leaderboard disabled:
screen shot 2018-02-05 at 2 55 33 pm

### CODE REVIEW

File: app/views/notifiers/course/level_notifier/reached/user_notifications/popup.json.jbuilder [hunk 1]
```diff
@@ -4,3 +4,15 @@ json.notificationType 'levelReached'
 
 level = notification.activity.object
 json.levelNumber level.level_number
+
+leaderboard_component = current_component_host[:course_leaderboard_component]
+leaderboard_enabled = leaderboard_component.present?
+json.leaderboardEnabled leaderboard_enabled
+if leaderboard_enabled
+  index = current_course.course_users.students.without_phantom_users.includes(:user).
```
Comment (weiqingtoh) [inline app/views/notifiers/course/level_notifier/reached/user_notifications/popup.json.jbuilder]: Can suggest to put this into a helper, and test that function?

File: app/views/notifiers/course/level_notifier/reached/user_notifications/popup.json.jbuilder [hunk 2]
```diff
@@ -4,3 +4,11 @@ json.notificationType 'levelReached'
 
 level = notification.activity.object
 json.levelNumber level.level_number
+
+leaderboard_component = current_component_host[:course_leaderboard_component]
+leaderboard_enabled = leaderboard_component.present?
+display_user_count = leaderboard_component.settings.display_user_count
```
Comment (weiqingtoh) [inline app/views/notifiers/course/level_notifier/reached/user_notifications/popup.json.jbuilder]: Sorry just to check if `leaderboard_component` is `nil` then this line will not work right? maybe can do something like:
```ruby
leaderboard_component = ...
if leaderboard_component
  display_user_count = ...
  ranking = ...
end
```

Comment (kxmbrian) [reply app/views/notifiers/course/level_notifier/reached/user_notifications/popup.json.jbuilder]: Thanks. Fixed and added test!

File: app/helpers/course/leaderboards_helper.rb [hunk 3]
```diff
@@ -6,4 +6,18 @@ module Course::LeaderboardsHelper
...
+  #
+  # @param [Course] course
+  # @param [CourseUser] course_user The student to query for.
+  # @param [Integer] display_user_count The number of positions available on the leaderboard
+  # @return [nil] if student is not on the leaderboard
+  # @return [Integer] position of the student on the leaderboard
+  def leaderboard_position(course, course_user, display_user_count)
+    index = course.course_users.students.without_phantom_users.includes(:user).
+            ordered_by_experience_points.take(display_user_count).
+            find_index { |student| student == course_user }
```
Comment (jeremyyap) [inline app/helpers/course/leaderboards_helper.rb]: Does `find_index(course_user)` work here? Other than that LGTM

Comment (kxmbrian) [reply app/helpers/course/leaderboards_helper.rb]: That's better. Thanks!

**Pull request merged**